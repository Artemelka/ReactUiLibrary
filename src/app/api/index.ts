import { getLabels, getLocales } from './localization';

export const API = {
    localization: {
        getLabels,
        getLocales
    }
};
