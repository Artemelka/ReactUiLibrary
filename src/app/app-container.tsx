import React, { Component } from 'react';
import thunk from 'redux-thunk';
import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { routerMiddleware } from 'react-router-redux';
import { reducers } from './reducers';
import { TranslateProvider, translateMiddleware } from '../services/translate';
import { request } from '../services';
import { App } from './index';

const history = createBrowserHistory();
const store = createStore(reducers, composeWithDevTools(
    applyMiddleware(
        routerMiddleware(history),
        translateMiddleware,
        thunk
)));

export class AppContainer extends Component {
    fetchDictionary = () => request('/api/translate');

    render() {
        return (
            <Provider store={store}>
                <TranslateProvider fetchDictionary={this.fetchDictionary}>
                    <Router history={history}>
                        <App />
                    </Router>
                </TranslateProvider>
            </Provider>
        );
    }
}
