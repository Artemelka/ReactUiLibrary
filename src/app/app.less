.App {
    bottom: 0;
    //font-family: 'Fira Sans', sans-serif;
    left: 0;
    overflow: auto;
    position: fixed;
    right: 0;
    top: 0;
}
