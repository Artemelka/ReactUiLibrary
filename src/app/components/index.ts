export {
    AppLoader,
    APP_LOADER_REDUCER_KEY,
    appLoaderIsLoadingSelector,
    appLoaderReducer,
    changeAppLoaderState
} from './app-loader';
export { CodeBlock } from './code-block/CodeBlock';
export { EditorForm } from './editor-form/EditorForm';
export { FooterAside } from './footer/FooterAside';
export { LanguageSelect } from './language-select';
export { SandboxPanel, SandboxPropsTable } from './sandbox/components';
export { LibrarySideBar } from './side-bar/LibrarySideBar';
export { SandboxLayout } from './sandbox';
export {
    FirstDialog,
    SecondDialog,
    ThirdDialog,
    ModalDialogName,
    FirstPanel,
    SecondPanel,
    ThirdPanel,
    ModalPanelName
} from './modals';
export { TranslateEditor } from './translate-editor';
export {
    TextArea,
    TextInput
} from './inputs';
