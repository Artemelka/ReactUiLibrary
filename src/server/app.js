import express from 'express';
import bodyParser from 'body-parser';
import fileUpload from 'express-fileupload';
import reload from 'express-reload';
import mongoose from 'mongoose';
import { DB_URL, DB_URL_CUT, ServerParams, KEEP_ALIVE_TIMEOUT } from './constants';

const app = express();
const { PORT, HOST } = ServerParams;
const path = __dirname + '/routes/';
const listenCallback = error => error
    ? console.log('something bad happened', error)
    : console.log(`Listening on port ${PORT}!`);

app.use(bodyParser.json());
app.use(bodyParser.text());
app.use(fileUpload());
app.use(express.static('dist'));
app.use('/api', reload(path));

const appServer = app.listen(PORT, HOST, listenCallback);

mongoose.connect(DB_URL, {
    useNewUrlParser: true,
    useUnifiedTopology: true
})
    .then(() => console.log('Connected to DB!'))
    .catch(error => console.error(error));

appServer.keepAliveTimeout = KEEP_ALIVE_TIMEOUT;
