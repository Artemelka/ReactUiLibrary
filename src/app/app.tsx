import * as React from 'react';
import { Route } from 'react-router';
import { TestHomePage, LibraryPage } from './pages';

import './app.less';

export class App extends React.Component<{}, {}> {
    render() {
        return (
            <div className="App">
                <Route path="/" exact component={TestHomePage} />
                {/*<Route path="/" exact component={LibraryPage} />*/}
                {/*<Route path="/library" component={LibraryPage} />*/}
            </div>
        );
    }
}

