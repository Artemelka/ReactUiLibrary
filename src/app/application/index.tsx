import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { AppLoader } from 'components';
import { StoreInjectorProvider } from '@artemelka/redux-store-injector';
import { appStore, history } from './redux';
import { AppContainer } from './App';
import { mockAllRequest } from '../api';

export const App = () => {
    if (process.env && process.env.MOCK) {
        mockAllRequest();
    }

    return (
        <Provider store={appStore}>
            <StoreInjectorProvider store={appStore}>
                <ConnectedRouter history={history}>
                    <AppContainer />
                    <AppLoader />
                </ConnectedRouter>
            </StoreInjectorProvider>
        </Provider>
    );
};
