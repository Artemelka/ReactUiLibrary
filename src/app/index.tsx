import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { appStore } from './store';
import { history } from './app-history';
import { AppContainer } from './app';
import { AppLoader } from './components';
import { mockAllRequest } from './api';

export const App = () => {
    if (process.env && process.env.MOCK) {
        mockAllRequest();
    }

    return (
        <Provider store={appStore}>
            <ConnectedRouter history={history}>
                <AppContainer />
                <AppLoader />
            </ConnectedRouter>
        </Provider>
    );
};
