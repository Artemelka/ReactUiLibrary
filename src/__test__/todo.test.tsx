import * as React from 'react';
import { shallow } from 'enzyme';

import { ToDo } from '../app';

describe('Test ToDo component', () => {
    test('Rendering snapshot', () => {
        const testTitle = ['test', 'test2'];
        const output = shallow(<ToDo title={testTitle} />);

        expect(output).toMatchSnapshot();
    });
});
