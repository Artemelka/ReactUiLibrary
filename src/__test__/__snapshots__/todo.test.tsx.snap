// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ToDo component Rendering snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToDo
    title={
      Array [
        "test",
        "test2",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          test
        </h1>,
        <h1>
          test2
        </h1>,
      ],
      "className": "Todo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "test2",
        },
        "ref": null,
        "rendered": "test2",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            test
          </h1>,
          <h1>
            test2
          </h1>,
        ],
        "className": "Todo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "test2",
          },
          "ref": null,
          "rendered": "test2",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
